packages/hardhat/contracts/periphery/ControllerHelper.sol

```diff
@@ -295,178 +131,243 @@ contract ControllerHelper is UniswapControllerHelper, AaveControllerHelper, IERC
     /**
      * @notice close short position with user Uniswap v3 LP NFT
      * @dev user should approve this contract for Uni NFT transfer
+     * @param _params ControllerHelperDataType.CloseShortWithUserNftParams struct
      */
-    function closeShortWithUserNft(CloseShortWithUserNftParams calldata _params) external {
+    function closeShortWithUserNft(ControllerHelperDataType.CloseShortWithUserNftParams calldata _params) external {
+        if (_params.vaultId != 0) require(IShortPowerPerp(shortPowerPerp).ownerOf(_params.vaultId) == msg.sender);
+
         INonfungiblePositionManager(nonfungiblePositionManager).safeTransferFrom(
             msg.sender,
             address(this),
             _params.tokenId
         );
 
-        _closeUniLp(
-            closeUniLpParams({
-                vaultId: _params.vaultId,
+        // close LP position
+        (uint256 wPowerPerpAmountInLp, ) = ControllerHelperUtil.closeUniLp(
+            nonfungiblePositionManager,
+            ControllerHelperDataType.closeUniLpParams({
                 tokenId: _params.tokenId,
                 liquidity: _params.liquidity,
```

Comment:
Do we need to specify liquidity in params or just grab from position manger:

(, , , , , , , uint128 liquidity, , , , ) INonFungiblePositionManager.positions(tokenId)

Reply:
Specify it in params to allow partial close